src/main/java/fun/epak/pak/service/UserService.java

```diff
@@ -71,4 +73,25 @@ private String getFileExtension(MultipartFile multipartFile) {
         return Objects.requireNonNull(multipartFile.getContentType())
                 .replaceFirst("image/", "");
     }
+
+    public UserProfileData loadUserProfileData(String email) {
+        User user = userRepository.findByEmail(email).orElseThrow();
+        String userImagePath = "/data/images/profiles/" + user.getId() + "/" + user.getImageName();
+        return UserProfileData.of(user, userImagePath);
+    }
+
+    public void saveChangedUser(ChangeUserDataRequest userData, MultipartFile multipartFile, String email) {
+        User user = userRepository.findByEmail(email).orElseThrow();
+        if (userData.getUsername() != null) {
+            user.setUsername(userData.getUsername());
```

Comment:
Chcemy móc zmienić username?

Reply:
tak, to nie email który jest unique, nie jest też żadnym kluczem obcym